articles/azure-app-configuration/enable-dynamic-configuration-dotnet-core-push-refresh.md

```diff
@@ -168,9 +164,7 @@ namespace TestConsole
 }
 ```
 
-The [SetDirty](/dotnet/api/microsoft.extensions.configuration.azureappconfiguration.iconfigurationrefresher.setdirty) method is used to set the cached value for key-values registered for refresh as dirty. This ensures that the next call to `RefreshAsync` or `TryRefreshAsync` re-validates the cached values with App Configuration and updates them if needed.
-
-A random delay is added before the cached value is marked as dirty to reduce potential throttling in case multiple instances refresh at the same time. The default maximum delay before the cached value is marked as dirty is 30 seconds, but can be overridden by passing an optional `TimeSpan` parameter to the `SetDirty` method.
+Ensuring that users get their most up to date configurations is nearly impossible as a configuration can change at any moment (even after a request is sent out). The `ProcessPushNotification` method incorporates a synchronization token into the next request to the `App Configuration`. This `Synchronization token` guarantees users receive all changes at least up to the point which the notification was triggered. There still is no guarantee that changes occuring after a push notification was triggered will be reflected in the response.
```

Comment:
This still talks too much about implementation details rather than explaining what the code does for the customer. I drafted something below and hope it makes sense to you.

----

The `ProcessPushNotification` method resets the cache expiration to a short random delay, so later calls to `RefreshAsync` or `TryRefreshAsync` will re-validate the cached values against App Configuration and update them as necessary. In this example, you register to monitor the key _TestApp:Settings:Message_ for change detection with a cache expiration time of one day, which means no requests to App Configuration will be made before a day has passed since the last check. By calling `ProcessPushNotification`, your application will send requests to App Configuration in the next few seconds. It will load new configuration values shortly after changes occur in App Configuration, but without polling for the changes all the time. In case of your application misses the change notification for any reason, it will still check for configuration changes once a day.

The short random delay for cache expiration is helpful if you have many instances of your application or microservices connecting to the same App Configuration store with the push model setup. Without it, all instances of your application could send requests to your App Configuration store simultaneously as soon as they receive a change notification, which could cause throttling to your store. This delay is a random number between 0 and a maximum of 30 seconds by default, but you can change it through the optional parameter `maxDelay` to the `ProcessPushNotification` method.

The `ProcessPushNotification` method takes in a `PushNotification` object containing information about which change in App Configuration triggered the push notification. It helps ensure all configuration changes up to the triggering event are loaded in the following configuration refresh. The `SetDirty` method does not guarantee the change that triggers the push notification to be loaded in an immediate configuration refresh. If you are using the `SetDirty` method for the push model, we recommend using the `ProcessPushNotification` method instead.

Reply:
Incorporated changes. Here are my changes to the second block of text:

The short random delay for cache expiration is helpful if you have many instances of your application or microservices connecting to the same App Configuration store with the push model. Without this delay, all instances of your application could send requests to your App Configuration store simultaneously as soon as they receive a change notification. This can cause the App Configuration Service to throttle your store. Cache expiration delay is set to a random number between 0 and a maximum of 30 seconds by default, but you can change the maximum value through the optional parameter `maxDelay` to the `ProcessPushNotification` method.